modulus/sym/eq/fd/grads.py

```diff
@@ -0,0 +1,239 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from typing import Union, List
+
+Tensor = torch.Tensor
+
+
+class FirstDerivO2(torch.nn.Module):
+    """Module to compute first derivative with 2nd order accuracy"""
+
+    def __init__(self, dim: int, dx: Union[float, List[float]]):
+        super().__init__()
+        self.dim = dim
+        if isinstance(dx, float):
+            dx = [dx for _ in range(dim)]
+        self.dx = dx
+
+        self.register_buffer("ddx1D", torch.Tensor([-1.0 / 2.0, 0.0, 1.0 / 2.0]))
```

Comment:
If we get to this point I would highly recommend writing warp kernels for these. We can have the same factory functionality and it will give us ~4x speed increase. Not needed for this PR though.

Reply:
Thanks! yes I agree that higher orders are needed, but I wanted to prioritize the structure in this PR. I have made this issue to keep a track of this: https://github.com/NVIDIA/modulus-sym/issues/180

I am marking this resolved currently :)